src/sms-verification/scripts/directives.js

```diff
@@ -0,0 +1,64 @@
+(function (angular) {
+    'use strict';
+
+    var module = angular.module('user_management.sms-verification');
+
+    module.directive('smsVerify', [
+        '$parse',
+        '$routeParams',
+        'smsVerificationFactory',
+        function ($parse, $routeParams, smsVerificationFactory) {
+            return {
+                restrict: 'A',
+                scope: true,
+                templateUrl: 'templates/user_management/sms-verification/verify_form.html',
+                link: function (scope, element, attrs) {
+
+                    // 2-way bind data attribute to scope.
+                    // scope: { data: '=' }
+                    if (angular.isDefined(attrs.data)) {
+                        // Set scope.data form attrs.data.
+                        scope.$watch(attrs.data, function(data) {
+                            scope.data = data;
+                        });
+                        // Set attrs.data from scope.data.
+                        var parentAssign = $parse(attrs.data).assign;
+                        scope.$watch('data', function(data) {
+                            parentAssign(scope.$parent, data)
+                        });
+                    }
+
+                    scope.verify = function () {
+                        scope.verified = false;
+                        smsVerificationFactory
+                            .verify.post(scope.data.activation_code, scope.data.phone_number)
+                            .then(function (response) {
+                                scope.status = response.status;
+                                scope.verified = true;
+                            }, function (response) {
```

Comment:
Another pattern I've seen people recommending is to use separate `.then()` and `.catch()` functions instead of the 2 parameter `(success, error)` form of `.then()`. 

It seems to be just a readability argument, but would change this to:

```
.then(function (response) {
    scope.status = response.status;
    scope.verified = true;
})
.catch(function (response) {
    scope.status = response.status;
    scope.errors = response.data;

    if (response.status === 400 && response.data.detail === 'Invalid or expired code.') {
        scope.invalid = true;
    }
});
```

I'm not sure if that's a useful thing to be doing or not...